src/controllers/controller.bar.js

```diff
@@ -322,7 +343,7 @@ module.exports = function(Chart) {
 
 			if (length === 1) {
 				leftSampleSize = base > start ? base - start : end - base;
-				rightSampleSize = base < end ? end - base : base - start;
+				rightSampleSize = leftSampleSize;
```

Comment:
I'm curious, which part of the issue does this fix?

Reply:
This code is part of a special case for `length === 1`, in the other cases where `index === 0` or `index === length - 1`, which is always the case, both sample sizes are set to the same value.